store/v2/root/snapshot_test.go

```diff
@@ -190,6 +190,12 @@ func TestMultistoreSnapshotRestore(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, sourceSubStore, targetSubStore)
 	}
+
+	// restore checking for non-empty store
+	target2 := newMultiStoreWithBasicData(t, memdb.NewDB(), 0)
```

Comment:
This does the same as above. We need to check that the restore fails if there is an existing data or schema in the target.
It would also be good to check that it fails if there's an existing saved version, even if it contains no data